cirq/ops/pauli_sum_exponential.py

```diff
@@ -0,0 +1,126 @@
+# Copyright 2020 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import Any, Iterator, Tuple, Union, TYPE_CHECKING
+
+import numpy as np
+import sympy
+
+from cirq import linalg, protocols, value
+from cirq.ops import linear_combinations, pauli_string_phasor
+
+if TYPE_CHECKING:
+    import cirq
+
+
+def _all_pauli_strings_commute(pauli_sum: 'cirq.PauliSum') -> bool:
+    for x in pauli_sum:
+        for y in pauli_sum:
+            if not protocols.commutes(x, y):
+                return False
+    return True
+
+
+@value.value_equality(approximate=True)
+class PauliSumExponential:
+    """Represents an operator defined by the exponential of a PauliSum.
+
+    Given a hermitian/anti-hermitian PauliSum PS_1 + PS_2 + ... + PS_N, this
+    class returns an operation which is equivalent to
+    exp(j * exponent * (PS_1 + PS_2 + ... + PS_N)).
+
+    This class only supports commuting Pauli terms.
+    """
+
+    def __init__(
+        self, pauli_sum_like: 'cirq.PauliSumLike', exponent: Union[int, float, sympy.Basic] = 1
+    ):
+        pauli_sum = linear_combinations.PauliSum.wrap(pauli_sum_like)
+        if not _all_pauli_strings_commute(pauli_sum):
+            raise ValueError("PauliSumExponential defined only for commuting pauli sums.")
+        self._multiplier = None
+        for pauli_string in pauli_sum:
+            coeff = pauli_string.coefficient
+            curr_multiplier = -1j if abs(coeff.imag) > 1e-8 else 1.0
```

Comment:
We typically extract this into a `atol` parameter with a default value of `1e-8`. 
```suggestion
            curr_multiplier = -1j if abs(coeff.imag) > atol else 1.0
```

Reply:
done